resin/models/supervisor.py

```diff
@@ -0,0 +1,363 @@
+                '{0}/update'.format(self.SUPERVISOR_API_VERSION),
+                'POST',
+                data=data,
+                endpoint=self.RESIN_SUPERVISOR_ADDRESS,
+                api_key=self.RESIN_SUPERVISOR_API_KEY
+            )
+
+    def reboot(self, device_id=None, app_id=None):
+        """
+        Reboot the device.
+        No need to set device_id and app_id if command is sent to the API on device.
+
+        Args:
+            device_id (Optional[str]): device id.
+            app_id (Optional[str]): application id.
+
+        Returns:
+            dict: when successful, this dictionary is returned `{ 'Data': 'OK', 'Error': '' }`.
+
+        Raises:
+            InvalidOption: if the endpoint is Resin API proxy endpoint and device_id or app_id is not specified.
+
+        """
+
+        if not self._on_device:
+            self._check_args(device_id, app_id)
+
+            data = {
+                'deviceId': device_id,
+                'appId': app_id
+            }
+
+            return self.base_request.request(
+                'supervisor/{0}/reboot'.format(self.SUPERVISOR_API_VERSION),
+                'POST',
+                data=data,
+                endpoint=self.settings.get('api_endpoint'),
+                login=True
+            )
+        else:
+            return self.base_request.request(
+                '{0}/reboot'.format(self.SUPERVISOR_API_VERSION),
+                'POST',
+                endpoint=self.RESIN_SUPERVISOR_ADDRESS,
+                api_key=self.RESIN_SUPERVISOR_API_KEY
+            )
+
+    def shutdown(self, device_id=None, app_id=None):
+        """
+        Shut down the device.
+        No need to set device_id and app_id if command is sent to the API on device.
+
+        Args:
+            device_id (Optional[str]): device id.
+            app_id (Optional[str]): application id.
+
+        Returns:
+            dict: when successful, this dictionary is returned `{ 'Data': 'OK', 'Error': '' }`.
+
+        Raises:
+            InvalidOption: if the endpoint is Resin API proxy endpoint and device_id or app_id is not specified.
+
+        """
+
+        if not self._on_device:
+            self._check_args(device_id, app_id)
+
+            data = {
+                'deviceId': device_id,
+                'appId': app_id
+            }
+
+            return self.base_request.request(
+                'supervisor/{0}/shutdown'.format(self.SUPERVISOR_API_VERSION),
+                'POST',
+                data=data,
+                endpoint=self.settings.get('api_endpoint'),
+                login=True
+            )
+        else:
+            return self.base_request.request(
+                '{0}/shutdown'.format(self.SUPERVISOR_API_VERSION),
+                'POST',
+                endpoint=self.RESIN_SUPERVISOR_ADDRESS,
+                api_key=self.RESIN_SUPERVISOR_API_KEY
+            )
+
+    def purge_data(self, app_id, device_id=None):
```

Comment:
Might it make sense to leave this just as `purge` so that the methods match the endpoint name? By now IMO we have it well documented that purge refers to /data.